packages/venia-concept/src/components/Checkout/container.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import { connect } from 'src/drivers';
+import { array, bool, func, object, oneOf, shape, string } from 'prop-types';
+
+import {
+    beginCheckout,
+    cancelCheckout,
+    editOrder,
+    submitShippingAddress,
+    submitOrder,
+    submitPaymentMethodAndBillingAddress,
+    submitShippingMethod
+} from 'src/actions/checkout';
+
+import Flow from './flow';
+
+const hasData = value => !!value;
+const isCartReady = cart => cart.details.items_count > 0;
+const isCheckoutReady = checkout => {
+    const {
+        billingAddress,
+        paymentData,
+        shippingAddress,
+        shippingMethod
+    } = checkout;
+
+    return [billingAddress, paymentData, shippingAddress, shippingMethod].every(
+        hasData
+    );
+};
+
+// TODO: This is exported for testing, but I think we should split this into a
+// separate file entirely and just do the prop/action shaping here in this file.
+export const CheckoutContainer = props => {
```

Comment:
Yea, I'm not even sure we need this component. We do the prop shaping, split it into multiple groups, and then [pull them apart](https://github.com/magento-research/pwa-studio/pull/1309/files#diff-c54588d5f498544c2028e38a1865dc0aR48) inside flow anyways.